File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    float letters = 0;
    float words = 1;
    float sentences = 0;

    string text = get_string("Text: ");

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] >= 'a' && text[i] <= 'z')
        {
            letters++;
        }
        if (text[i] >= 'A' && text[i] <= 'Z')
        {
            letters++;
        }
        if (text[i] == ' ')
        {
            words++;
        }
        if (text[i] == '.')
        {
            sentences++;
        }
        if (text[i] == '!')
        {
            sentences++;
        }
        if (text[i] == '?')
        {
            sentences++;
        }
    }

    float avgl = (letters / words) * 100;
    float avgs = (sentences / words) * 100;
    float grade = 0.0588 * avgl - 0.296 * avgs - 15.8;

    if (grade < 1 )
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(grade));
    }
}

File: pset6/readability/readability.py
from cs50 import get_string

letters = 0
words = 1
sentences = 0
text = get_string("Text: ")
n = len(text)

for i in range(0, n, 1):
    if text[i] >= 'a' and text[i] <= 'z':
        letters += 1

    if text[i] >= 'A' and text[i] <= 'Z':
        letters += 1

    if text[i] == ' ':
        words += 1

    if text[i] == '.':
        sentences += 1

    if text[i] == '!':
        sentences += 1

    if text[i] == '?':
        sentences += 1

avgl = (letters / words) * 100
avgs = (sentences / words) * 100
grade = 0.0588 * avgl - 0.296 * avgs - 15.8

if grade < 1:
    print("Before Grade 1")

elif grade >= 16:
    print("Grade 16+")

else:
    print("Grade ", + round(grade))

